lib/CPathfinder.h

```diff
@@ -121,4 +176,36 @@ class CPathfinder : private CGameInfoCallback
 	bool addTeleportOneWay(const CGTeleport * obj) const;
 	bool addTeleportOneWayRandom(const CGTeleport * obj) const;
 	bool addTeleportWhirlpool(const CGWhirlpool * obj) const;
+
+	bool canVisitObject() const;
+
+};
+
+struct TurnInfo
+{
+	int turn;
+	int maxMovePointsLand;
+	int maxMovePointsWater;
+	const Bonus * bonusFlying;
+	const Bonus * bonusWaterWalking;
+};
+
+class DLL_LINKAGE CPathfinderHelper
+{
+public:
+	TurnInfo * ti;
```

Comment:
Actually "hlp->ti->something" and "hlp->ti[turn]->something" are both bad design.